src/App.tsx

```diff
@@ -15,48 +21,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const goodsWithKeys :Good[] = goodsFromServer.map((elem, index) => {
+  return {
+    good: elem,
+    id: index,
+  };
+});
+
+enum SortType {
+  Length = 'Length',
+  Alphabetically = 'Alphabetically',
+}
+
+interface SortParams {
+  sortParam : string;
+  isReversed: boolean;
+}
+
+const getPreparedGoods = (
+  goods: Good[],
+  { sortParam, isReversed } :SortParams,
+) :Good[] => {
+  const prepareGoods = [...goods];
+
+  if (sortParam) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortParam) {
+        case SortType.Alphabetically:
+          return good1.good.localeCompare(good2.good);
+
+        case SortType.Length:
+          return good1.good.length - good2.good.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortParam, setSortParam] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const reparedGoods = getPreparedGoods(
+    goodsWithKeys,
+    { sortParam, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classes(
+            'button',
+            'is-info',
+            { 'is-light': sortParam !== SortType.Alphabetically },
+          )}
+          onClick={() => setSortParam(SortType.Alphabetically)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classes(
+            'button',
+            'is-success',
+            { 'is-light': sortParam !== SortType.Length },
+          )}
+          onClick={() => setSortParam(SortType.Length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classes(
+            'button',
+            'is-warning',
+            { 'is-light': !isReversed },
+          )}
+          onClick={() => setIsReversed(!isReversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortParam || isReversed)
+          && (
```

Comment:
```suggestion
        {(sortParam || isReversed) && (
```
Let's create a variable for this condition

Reply:
Thanks for good advices!